<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IOPage</class>
 <widget class="QWidget" name="IOPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>876</width>
    <height>404</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_4">
   <item>
    <widget class="QWidget" name="widgetController" native="true">
     <layout class="QGridLayout" name="gridLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="9" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Output control</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Text format</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <layout class="QVBoxLayout" name="verticalLayoutCommunicationController"/>
      </item>
      <item row="7" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="pushButtonCommunicationSettings">
          <property name="text">
           <string>⚙Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonCommunicationOpen">
          <property name="text">
           <string>🔗Open</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="17" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Cycle interval</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="15" column="0" colspan="2">
       <widget class="QLabel" name="label_3">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Input control</string>
        </property>
       </widget>
      </item>
      <item row="18" column="1">
       <widget class="QComboBox" name="comboBoxInputFormat"/>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="Line" name="line_4">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="16" column="0" colspan="2">
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QComboBox" name="comboBoxOutputFormat"/>
      </item>
      <item row="19" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="pushButtonInputSettings">
          <property name="text">
           <string>⚙Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonInputWriteBytes">
          <property name="text">
           <string>📤Send</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Communication control</string>
        </property>
       </widget>
      </item>
      <item row="18" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Text format</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QCheckBox" name="checkBoxOutputRx">
          <property name="text">
           <string>Rx</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="checkBoxOutputTx">
          <property name="text">
           <string>Tx</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="checkBoxOutputDate">
          <property name="text">
           <string>Date</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QCheckBox" name="checkBoxOutputFlag">
          <property name="text">
           <string>Flag</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="checkBoxOutputTime">
          <property name="text">
           <string>Time</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QCheckBox" name="checkBoxOutputMs">
          <property name="text">
           <string>Ms</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="17" column="1">
       <widget class="QComboBox" name="comboBoxInputInterval">
        <property name="toolTip">
         <string notr="true">ms</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButtonOutputSettings">
          <property name="text">
           <string>⚙Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonOutputClear">
          <property name="text">
           <string>🆑Clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QComboBox" name="comboBoxCommmunicationTypes"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_2" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLineEdit" name="lineEditInput">
          <property name="text">
           <string notr="true"/>
          </property>
          <property name="placeholderText">
           <string>Paoload ([Prefix][Payload(crc data)][CRC][Suffix])</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonInputPreset">
          <property name="text">
           <string notr="true">🗒</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="spacing">
         <number>10</number>
        </property>
        <item>
         <widget class="QLabel" name="labelTitle">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Output</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonOutput">
          <property name="toolTip">
           <string>Output</string>
          </property>
          <property name="text">
           <string notr="true">🔡</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonEmitter">
          <property name="toolTip">
           <string>Emiter</string>
          </property>
          <property name="text">
           <string>📡</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonResponser">
          <property name="toolTip">
           <string>Responser</string>
          </property>
          <property name="text">
           <string>🔃</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonPreset">
          <property name="toolTip">
           <string>Preset</string>
          </property>
          <property name="text">
           <string>🧾</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="labelRx">
          <property name="styleSheet">
           <string notr="true">color:blue</string>
          </property>
          <property name="text">
           <string notr="true">Rx</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="StatisticianUi" name="widgetRxInfo" native="true"/>
        </item>
        <item>
         <widget class="Line" name="lineRxTx">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelTx">
          <property name="styleSheet">
           <string notr="true">color:green</string>
          </property>
          <property name="text">
           <string notr="true">Tx</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="StatisticianUi" name="widgetTxInfo" native="true"/>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonUp">
          <property name="text">
           <string>⬆</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonDown">
          <property name="text">
           <string>⬇</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonLeft">
          <property name="text">
           <string>⬅</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonRight">
          <property name="text">
           <string>➡️</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QStackedWidget" name="stackedWidget">
        <widget class="QWidget" name="pageOutput">
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTextBrowser" name="textBrowserOutput"/>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="font">
               <font>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Input</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelInfo">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string notr="true"/>
              </property>
              <property name="alignment">
               <set>Qt::AlignmentFlag::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelCrc">
              <property name="text">
               <string notr="true"/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="xTools::PresetUi" name="pagePreset"/>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>StatisticianUi</class>
   <extends>QWidget</extends>
   <header location="global">IO/Ui/Processor/StatisticianUi.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>xTools::PresetUi</class>
   <extends>QWidget</extends>
   <header location="global">IO/UI/Model/PresetUi.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonOutputClear</sender>
   <signal>clicked()</signal>
   <receiver>textBrowserOutput</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>245</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
